src/htmlrenderer.cpp

```diff
@@ -1341,4 +1409,61 @@ std::string HtmlRenderer::format_ol_count(unsigned int count, char type)
 	}
 }
 
+LinkPair HtmlRenderer::replace_iframe(const LinkPair& iframe)
+{
+	/**
+	 * Check if the iframe's `src` attribute can be mapped to a more usable URL
+	 * using a known mapping. Each mapping is represented by a 3-tuple with the
```

Comment:
I think Newsboat shouldn't change the URLs, because this is brittle, opinionated, and is not configurable.

Brittle, because YouTube/Vimeo/Spotify might change their endpoints someday, and the URLs that Newsboat generates might become unusable. We can fix that, of course; we can even issue point releases for the last few minor releases to cover as many users as possible, but I'd rather not put ourselves in that position at all.

Opinionated, because I personally find YouTube embeds totally usable, perhaps even more so than the "full-featured" pages, where I'm distracted by recommendations and comments.

Not configurable — well, obviously it isn't :) The reason I can about this property is because I'm aware of alternative YouTube front-ends like Invidious, and once Newsboat starts rewriting links, why not add an option to rewrite youtube.com into invidio.us, right? And it's self-hosted, so we better add a setting for the domain to rewrite to… It's a rabbit hole, and I'd rather avoid it :)

I think the rewriting should happen outside of Newsboat. We already provide ways to pre-process the feed (`filter:` in the urls file) and to supply user's own urlview (`external-url-viewer` setting), so I think we cover this well enough.

If you don't have any counter-arguments to that, please remove this method.

Reply:
Thanks for this feedback. I agree with you and admit I got a little carried away here :sweat_smile: -- Newsboat's pre-processing capabilities via external scripts seems like the better place for this logic.